src/GlobalVariables.as

```diff
@@ -158,22 +160,24 @@ package
             sql_conn.addEventListener(SQLErrorEvent.ERROR, sql_errorHandler);
 
             sql_conn.openAsync(sql_db);
+        }
 
-            function sql_openHandler(event:SQLEvent):void
-            {
-                sql_connect = true;
+        private function sql_openHandler(event:SQLEvent):void
+        {
+            sql_connect = true;
 
-                trace("Database was connected successfully");
+            trace("Database was connected successfully");
 
-                SQLQueries.refreshStatements(sql_conn);
-            }
+            SQLQueries.refreshStatements(sql_conn);
+            this.dispatchEvent(new Event(DB_CONNECT_COMPLETE));
+        }
 
-            function sql_errorHandler(event:SQLErrorEvent):void
-            {
-                sql_connect = false;
-                trace("Error message:", event.error.message);
-                trace("Details:", event.error.details);
-            }
+        private function sql_errorHandler(event:SQLErrorEvent):void
+        {
+            sql_connect = false;
+            trace("Error message:", event.error.message);
+            trace("Details:", event.error.details);
+            this.dispatchEvent(new Event(DB_CONNECT_ERROR));
```

Comment:
Being a bit nitpicky here, but I would remove the event listeners when either of these functions are called as follows:

```
private function sql_openHandler(event:SQLEvent):void
{
    sql_conn.removeEventListener(SQLEvent.OPEN, sql_openHandler);
    sql_conn.removeEventListener(SQLErrorEvent.ERROR, sql_errorHandler);

    sql_connect = true;

    trace("Database was connected successfully");

    SQLQueries.refreshStatements(sql_conn);
    this.dispatchEvent(new Event(DB_CONNECT_COMPLETE));
}

private function sql_errorHandler(event:SQLErrorEvent):void
{
    sql_conn.removeEventListener(SQLEvent.OPEN, sql_openHandler);
    sql_conn.removeEventListener(SQLErrorEvent.ERROR, sql_errorHandler);

    sql_connect = false;
    trace("Error message:", event.error.message);
    trace("Details:", event.error.details);
    this.dispatchEvent(new Event(DB_CONNECT_ERROR));
}
```